system/jlib/jptree.cpp

```diff
@@ -7609,48 +7667,82 @@ static constexpr const char * currentVersion = "1.0";
 /*
  * Use source to overwrite any changes in target
  *   Attributes are replaced
- *   Elements with no name attribute are assumed to match a single element in the target.  They are added if not present.
- *   Elements with a name attribute are matched by name.  If there is a match and the source element has an attribute
- *     '__remove__' then that element is removed, otherwise it is merged.  If there is no match it is added.
+ *   Singleton elements are replaced.
+ *   Entire arrays of scalar elements are replaced.
+ *   Entire arrays of elements with no name attribute are replaced.
+ *   Elements with a name attribute are matched by name.  If there is a match it is merged.  If there is no match it is added.
 */
 
-void mergeConfiguration(IPropertyTree & target, IPropertyTree & source)
+static bool checkInSequence(IPropertyTree & child, StringAttr &seqname, bool &first, bool &endprior)
+{
+    first = false;
+    endprior = false;
+    if (seqname.length() && streq(seqname, child.queryName()))
+        return true;
+    endprior = !seqname.isEmpty();
+    if (child.isArray(nullptr))
+    {
+        first=true;
+        seqname.set(child.queryName());
+        return true;
+    }
+    seqname.clear();
+    return false;
+}
+
+inline bool isScalarItem(IPropertyTree &child)
+{
+    if (child.hasChildren())
+        return false;
+    return child.getAttributeCount()==0;
+}
+
+static IPropertyTree *ensureMergeConfigTarget(IPropertyTree &target, const char *tag, const char *name, bool sequence)
+{
+    StringBuffer tempPath;
+    const char * path = (sequence) ? nullptr : tag;
+    if (name)
+    {
+        tempPath.clear().append(tag).append("[@name=\'").append(name).append("']");
```

Comment:
Does @name need to be passed in e.g. if it actually matched altNameAttribute?

Reply:
agreed will change